File: plugin/shared/src/main/java/com/jonathanpotts/blockmaps/shared/models/WorldModel.java
package com.jonathanpotts.blockmaps.shared.models;

/**
 * Stores data needed to process a world.
 */
public class WorldModel {
  /**
   * The name of the world.
   */
  public String name;

  /**
   * The spawn point of the world.
   */
  public VectorXYZ spawn;

  /**
   * The minimum height of the world.
   */
  public int minHeight;

  /**
   * The maximum height of the world.
   */
  public int maxHeight;
}

File: web/src/models/WorldModel.ts
import VectorXYZ from './VectorXYZ';

/**
 * Stores data needed to process a world.
 */
interface WorldModel {
  /**
   * The name of the world.
   */
  name: string;

  /**
   * The spawn point of the world.
   */
  spawn: VectorXYZ;

  /**
   * The minimum height of the world.
   */
  minHeight: number;

  /**
   * The maximum height of the world.
   */
  maxHeight: number;
}

export default WorldModel;
